src/StringTools/WeakStringCache.Concurrent.cs

```diff
@@ -62,11 +63,15 @@ public string GetOrCreateEntry(ref InternableString internable, out bool cacheHi
 
             handle = new StringWeakHandle();
             handle.SetString(result);
-            _stringsByHashCode.TryAdd(hashCode, handle);
+            if (_stringsByHashCode.TryAdd(hashCode, handle))
+            {
+                Interlocked.Add(ref _count, 1);
```

Comment:
```suggestion
                Interlocked.Increment(ref _count);
```

Reply:
Btw. it seems that it's allowed to concurrently call `GetOrCreateEntry` and `Scavenge` - so it meens it's possible that couple threads add items to the dictionary (line 66), but beefore they managed to increment the counter, the Scavenge call sets the counter to the current number of items in the dict, and only then the adding threads proceed incrementing the counter - so it's artificially higher than it should be.

It's probably not a big deal, because it's not a correctness issue, but rather extra work issue. But then it's at least worth comment that clarifis the code is ok with such eventuality.

If we would want to prevent such case, then we'd need to add some double checking/locking strategy